### PR TITLE
Don't crash on bird csvs, add vulnerability column

### LINKED ISSUE #43: Make addCSV in Data fail on a process failure (i.e. uploading a csv file with no columns)

### LINKED ISSUE #61: handle bird csvs, specifically, their species

### LINKED ISSUE #149: Make a list of all columns we expect and make use of in the sidebar
This is a subtask for allowing bird csvs. Also, we could move some of them, such as date and time, to more left, so that the table can be sorted easily by date and time

**If doesn't have some of them, such as time, reject as well**

set to default values like 1960:01:01 or <empty>

### PR DESCRIPTION
**Pull request by CGDogan**
resolves #61 resolves #149 (I think, as this file contains all column names we expect in a nonempty database) resolves #43

Also resolves: Leonard's request (I don't think we're opened an issue for this) to fill empty cells with something, which I chose as "[empty]"

Screenshot 2024-02-28 at 16 48 57

The 0 is caused by data, we may leave it as is

- Process bird species and map to high frequency format
- trim cells
- label empty cells
- if minimum probability is 0, it doesn't need to check the table for probability

### CODE REVIEW

File: devfiles/src/classes/data/datautils/csvreader.test.ts [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    /*console.log*/ expect(parseCSV(`a,"a\n"\nx,"y\n"`, undefined));
+    /*console.log*/ expect(parseCSV(`a,"a,\nb"`, undefined));
+    /*console.log*/ expect(parseCSV(`a,"a\n,b"`, undefined));
+    /*console.log*/ expect(parseCSV(`a,"a,b"`, undefined));
+    /*console.log*/ expect(parseCSV(`a,b,c,d`, undefined));
+
+    /*expect(linkElement).toBeInTheDocument();*/
+});
+
+/* eslint no-undef: off */
```
Comment (George-Ogden) [inline devfiles/src/classes/data/datautils/csvreader.test.ts:38]: Can you get rid of these comments, please?

**Review by George-Ogden — COMMENTED** [review]
Let me know when this is ready to merge and I will review functionality.

**Comment by CGDogan** [comment]
I think it's ready - once tests pass due to rebase

**Comment by CGDogan** [comment]
done

**Review by Hex27 — APPROVED** [review]

**Pull request merged**